powerbiclient/authentication.py

```diff
@@ -104,27 +104,34 @@ def _acquire_token_device_code(self):
 class InteractiveLoginAuthentication(AuthenticationResult):
 
     # Methods
-    def __init__(self):
+    def __init__(self, tenant_id=None):
         """Acquire token interactively i.e. via a local browser
 
+        Args:
+            tenant_id (string): Optional.
+                Id of Power BI tenant where your report resides.
+                
         Returns:
             object: Interactive authentication object
         """
-        auth_result = self._acquire_token_interactive()
+        auth_result = self._acquire_token_interactive(tenant_id)
         super().__init__(auth_result)
 
-    def _acquire_token_interactive(self):
+    def _acquire_token_interactive(self, tenant_id=None):
         """Returns the authentication result captured using interactive login
 
         Returns:
             dict: authentication result
         """
-        app = msal.PublicClientApplication(client_id=CLIENT_ID)
+        if not tenant_id:
+            app = msal.PublicClientApplication(client_id=CLIENT_ID)
+        else:
+            app = msal.PublicClientApplication(client_id=CLIENT_ID, authority=f"https://login.microsoftonline.com/{tenant_id}")
```

Comment:
Please apply the changes also for device code login flow. In addition, add the authority template string in the top of the file to be reused for both flows.